### DISCUSSION TITLE
Doorbell node with an output in Node Red?

### DISCUSSION THREAD

**Discussion by BruceCrowthorne** [andrei-tatar/node-red-contrib-smartnora#288] [Q&A]
I am a beginner so be gentle with me!

I have done my first project with Smart Nora. Using the Notify node I have the icon in my Google Home App and when I click the button it publishes an MQTT message. That is subscribed to by an 8266 (Wemos D1 Mini) with a relay. This switches on my central heating for 20 minutes as a "boost" or I can switch it on on my way home.  BRILLIANT ! Smart Nora is a great little utility!

So my next project is to add a remote sounder to my doorbell. I have a couple of Google smart speakers and that gets me so far, but what I would like is to publish an MQTT message when the doorbell is pressed. That will be used to sound a buzzer in my garage!

It appears that the "doorbell" node will detect the doorbell press, but that node doesn't have an output in Node Red.

My Node.js skills aren't good enough to modify the existing node.
Is there an alternative node that I could use?

Thanks and regards
Bruce

**Comment by andrei-tatar** [comment]
@BruceCrowthorne The doorbell node is intended as a way to notify your Google Home devices when someone is at the door. It does seem to match your use, if you want to be notified on the Google Home device you have in your garage.

In this case, there's no need for an output on the node, since you are telling it when to notify by injecting a message (and it can't be triggered from Google Home app anyways). There's also an example flow attached in the documentation of the node: https://github.com/andrei-tatar/node-red-contrib-smartnora/tree/master/doc/nodes/doorbell

In your case, you would need to change the `inject` node with something that comes from `mqtt`.

Demo: https://www.youtube.com/shorts/pbyahrdAh1c

**Reply by BruceCrowthorne** [reply]
Hi Andrei, thanks for your very prompt answer!

Just to clarify, the device in the garage isn't a Google Home device. It is a Wemos D1 Mini listening on MQTT and will use a buzzer as it's output.
Hence the need for an output from the doorbell node to feed to an MQTT node.

I hope that the attached image makes it clearer!
 
It shows how I have done my heating system, the MQTT message is published to a public broker and a Wemos D1 Mini subscribes and it's action is to switch on a relay for 20 minutes.

What I would like is to use another MQTT message which would go to another Wemos D1 Mini and that will sound a buzzer.

![image](https://github.com/andrei-tatar/node-red-contrib-smartnora/assets/17990508/1f961a60-ac2b-487b-b2b2-b4e48a1de8e4)

Hope that clarifies what I am trying to do
Thanks
Bruce

**Reply by andrei-tatar** [reply]
@BruceCrowthorne in that case I don't think the doorbell is the right node for you. It's meant as a "notify via Google home", not as a button. In fact you can see there's no way to trigger it from the Google Home app (which is not under Nora's control).

I suspect you would want a switch node.

**Reply by BruceCrowthorne** [reply]
Thanks Andrei, I have got it going with a Switch node.

Basically I created a Google action using the doorbell as a starter, the Switch as the action. I take the output from the switch and feed that to an MQTT Out node and publish a message. I then have a Wemos D1 Mini with a little buzzer which listens to the MQTT message as my sounder.

Great Node - I guess it is time to subscribe!!!!
